pkg/controller/quota/quota.go

```diff
@@ -53,7 +54,7 @@ func WaitForAllocation(req router.Request, resp router.Response) error {
 	*/
 	if cond := quotaRequest.Status.Condition(adminv1.QuotaRequestCondition); cond.Error {
 		status.Error(fmt.Errorf("quota allocation failed: %v", cond.Message))
-	} else if err != nil || !quotaRequest.Spec.Resources.Equals(quotaRequest.Status.AllocatedResources) {
+	} else if err != nil || waitingForAllocation(quotaRequest, appInstance) {
```

Comment:
nit: the `err != nil` check is no longer needed because the generation check will detect when the quota request wasn't found.